packages/hardhat-core/src/internal/core/providers/http.ts

```diff
@@ -86,6 +86,15 @@ export class HttpProvider extends EventEmitter implements EIP1193Provider {
   }
 
   public async request(args: RequestArguments): Promise<unknown> {
+    if (args.method === "hardhat_setLedgerOutputEnabled") {
```

Comment:
The only thing I'd change is adding a comment explaining why this is here and that it's a temporary fix.

Reply:
Good point, adding ➕ 

```suggestion
    // This is a temporary fix to an issue with noisy warnings in the logs
    // of a local node (#5406). This will be fixed in the next major release.
    if (args.method === "hardhat_setLedgerOutputEnabled") {
```